lib/core/src/Cardano/Wallet/Primitive/Types.hs

```diff
@@ -1870,3 +1889,49 @@ invariant msg a predicate =
 distance :: (Ord a, Num a) => a -> a -> a
 distance a b =
     if a < b then b - a else a - b
+
+data PoolMetadataSource
+    = None
+    | FetchDirect
+    | FetchSMASH URI
+    deriving (Show, Generic, Eq)
+
+instance ToText PoolMetadataSource where
+    toText None = (T.pack "")
+    toText FetchDirect = (T.pack "direct")
+    toText (FetchSMASH uri) = T.pack $ uriToString id uri ""
+
+instance FromText PoolMetadataSource where
+    fromText "" = Right None
```

Comment:
What about making this explicit as `"none"` ?

Reply:
Done